pkg/infra/baseagent.go

```diff
@@ -213,6 +215,31 @@ func (a *Base) HealthInfo() *health.HealthInfo {
 	return a.tower.healthInfo
 }
 
+func (a *Base) ActiveTreaties() map[uuid.UUID]messages.Treaty {
+	return a.activeTreaties
+}
+
+func (a *Base) AddToActiveTreaties(treaty messages.Treaty) {
+	treatyID := treaty.Id()
+	a.activeTreaties[treatyID] = treaty
+}
+
+func (a *Base) DeleteFromActiveTreaties(treaty messages.Treaty) {
+	treatyID := treaty.Id()
+	delete(a.activeTreaties, treatyID)
+}
```

Comment:
```suggestion
func (a *Base) DeleteTreaty(treatyID uuid.UUID) {
	delete(a.activeTreaties, treatyID)
}
```

Reply:
Would rather pass treaty as argument, but otherwise fine with this